### ISSUE TITLE
exp10(Dec128(0)) produces incorrect result

### ISSUE THREAD

**Issue by jmkuhn** [JuliaMath/DecFP.jl#47]
```
julia> exp10(Dec32(0))
+1E+0

julia> exp10(Dec64(0))
+1E+0

julia> exp10(Dec128(0))
+0E+0
```

**Comment by stevengj** [comment]
Oh weird, I see the same thing!  This looks like it could be a bug in the underlying library.

**Comment by stevengj** [comment]
Yup, the source code `bid128_exp10.c` contains the lines:
```c
    // x is 0, return 1.0
	 res.w[BID_HIGH_128W] = 0x3040000000000000ull;
	 res.w[BID_LOW_128W] = 0;
     BID_RETURN (res);
```
whereas I think it should be `res.w[BID_LOW_128W] = 1`.

**Comment by stevengj** [comment]
Even simpler, the code for `bid128_exp2.c` uses `res = BID128_1;` for the same condition, since `BID128_1` is apparently a predefined constant for 1.0.

**Comment by stevengj** [comment]
Intel responded to confirm the problem:
> Thank you for pointing this out. Indeed, this is a corner case that requires a correction…
I believe your change is correct (I will double-check and will confirm). We will also post an updated library in the coming days.

**Comment by quinnj** [comment]
@stevengj, do you know if the library was updated? Do we need to do anything to build the new version?

**Comment by stevengj** [comment]
I don't see any updates on Intel's website; I pinged them again via email to see what the timeframe is.

**Comment by stevengj** [comment]
I just got an email from them that they plan a new release "within days."

**Comment by quinnj** [comment]
@stevengj, do you have an official diff/patch for the fix? I have the BinaryBuilder/BinaryProvider working for all platforms now, so we could apply the fix ourselves until Intel gets around to releasing the new version.

**Comment by stevengj** [comment]
I have an unofficial copy of version 20U2, but I'm not sure if I'm supposed to distribute it.  They are supposed to post it within a week.   In the meantime, you can just use my one-line patch `res.w[BID_LOW_128W] = 1` from above.

**Comment by stevengj** [comment]
Version 20U2 was just officially released: http://www.netlib.org/misc/intel/IntelRDFPMathLib20U2.tar.gz

**Comment by jmkuhn** [comment]
In April I submitted a PR to DecFPBuilder https://github.com/quinnj/DecFPBuilder/pull/6 for updates to BinaryBuilder.  I'm not sure if there have been any significant changes to BinaryBuilder since then.

**Comment by jmkuhn** [comment]
Fixed with #106
